File: JavaScript/abc021A.js
function Main(input) {
    var a = input.split("\n");
    var n = parseInt(a[0],10);
    var c = [ 8, 4, 2, 1 ]
    var l = [];
    var i = 0;
    var j = 0;
    while(n != 0){
      if(c[j] <= n){
        n = n - c[j];
        l[i] = c[j];
        i++;
      }
      j++;
    }
    j = 0;
    console.log(i);
    while(j < i){
      console.log(l[j]);
      j++;
    }
  }
  
  Main(require("fs").readFileSync("/dev/stdin", "utf8"));
  

File: Python/abc021A.py
l = []
c = [8,4,2,1]
i = 0
n = int(input())
for a in c:
    if(n == 0):
      break
    if(a <= n):
      n = n - a
      l.append(a)
      i = i + 1
print(i)
for b in l:
    print(b)
